sycl/include/CL/sycl/accessor.hpp

```diff
@@ -514,10 +519,111 @@ class image_accessor
         "Read API is not implemented on host.");
 #endif
   }
+};
+
+template <typename DataT, int Dimensions, access::mode AccessMode,
+          access::placeholder IsPlaceholder>
+class __image_slice__ {
```

Comment:
I think this class should contain more methods according to the SYCL specification.
"`__image_array_slice__` is  an unspecified type which provides the interface of `accessor< dataT, dimensions, mode, access::target::image, access::placeholder:: false_>`"
Can you add TODO for `__image_array_slice__ operator[](size_t index) const`.

Reply:
operator[] work only for accessors with access::target::image_array but __image_array_slice__ provides interface of accessor with access::target_image.